lib/util/stream_utils.js

```diff
@@ -347,19 +347,33 @@ shaka.util.StreamUtils = class {
     await shaka.util.StreamUtils.getDecodingInfosForVariants(
         manifest.variants, usePersistentLicenses, /* srcEquals= */ false,
         /** preferredKeySystems= */ []);
+
+    const ContentType = shaka.util.ManifestParserUtils.ContentType;
+    const Capabilities = shaka.media.Capabilities;
+    const ManifestParserUtils = shaka.util.ManifestParserUtils;
+    const MimeUtils = shaka.util.MimeUtils;
+    const StreamUtils = shaka.util.StreamUtils;
+    const isXboxOne = shaka.util.Platform.isXboxOne();
+
     manifest.variants = manifest.variants.filter((variant) => {
       // See: https://github.com/shaka-project/shaka-player/issues/3860
       const video = variant.video;
+      const videoWidth = video.width || 0;
```

Comment:
Ah, now I see something I had overlooked before.  video could be null.  You can use `video?.width || 0`.  At least, I think the "optional chaining" operator is allowed by the compiler.  Please try that, here and for videoHeight.

The rest looks great!

Reply:
unfortunately is doesn't seem to do that - I updated the variables without using optional chaining